### PR TITLE
refactor(kube-api-rewriter): use unique renames for each apiGroup

### PR DESCRIPTION
**Pull request by diafour**
## Description

Remove single RenamedGroup field, add Renamed field into GroupRule.

```
Before:

cdi.kubevirt.io     \
kubevirt.io           \
clone.kubevirt.io       \
export.kubevirt.io        -> internal.virtualization.deckhouse.io
instancetype.kubevirt.io /
migrations.kubevirt.io  /
pool.kubevirt.io       /
snapshot.kubevirt.io  /

Now:

cdi.kubevirt.io  ->  cdi.internal.virtualization.deckhouse.io
kubevirt.io  ->  internal.virtualization.deckhouse.io
clone.kubevirt.io  ->  clone.internal.virtualization.deckhouse.io
export.kubevirt.io  ->  export.internal.virtualization.deckhouse.io
instancetype.kubevirt.io  ->  instancetype.internal.virtualization.deckhouse.io
migrations.kubevirt.io  ->  migrations.internal.virtualization.deckhouse.io
pool.kubevirt.io  ->  pool.internal.virtualization.deckhouse.io
snapshot.kubevirt.io  ->  snapshot.internal.virtualization.deckhouse.io
```

## Why do we need it, and what problem does it solve?

Simplify restore process by replacing one-to-many relation with one-to-one.

- Simplify discovery rewrite.
- No ambiguity in restoring star rules for roles. (Proper fix instead of the workaround PR #209)
- Get rid of origGroup argument in restore methods.
- APIGroup discovery response is now restored, not re-created.

## What is the expected result?

Each apiGroup has unique renamed name.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/kube-api-proxy/pkg/rewriter/affinity.go [hunk 1]
```diff
@@ -87,9 +92,20 @@ func rewriteNodeSelectorTerm(rules *RewriteRules, obj []byte, action Action) ([]
...
 func rewriteSelectorRequirement(rules *RewriteRules, obj []byte, action Action) ([]byte, error) {
-	return TransformString(obj, "key", func(field string) string {
-		return rules.LabelsRewriter().Rewrite(field, action)
-	})
+	key := gjson.GetBytes(obj, "key").String()
+	valuesArr := gjson.GetBytes(obj, "values").Array()
+	values := make([]string, 0, len(valuesArr))
+	for _, value := range valuesArr {
+		values = append(values, value.String())
+	}
```
Comment (yaroslavborbat) [inline images/kube-api-proxy/pkg/rewriter/affinity.go:100]:
```suggestion
	values := make([]string, len(valuesArr))
	for i, value := range valuesArr {
		values[i] = value.String()
	}
```

File: images/kube-api-proxy/pkg/rewriter/list.go [hunk 2]
```diff
@@ -58,6 +62,9 @@ func RewriteArray(obj []byte, arrayPath string, transformFn func(item []byte) ([
 	for _, item := range items {
 		rwrItem, err := transformFn([]byte(item.Raw))
 		if err != nil {
+			if err == SkipItem {
+				continue
+			}
```
Comment (yaroslavborbat) [inline images/kube-api-proxy/pkg/rewriter/list.go:67]:
```suggestion
			if errors.Is(err, SkipItem) {
				continue
			}
```

**Review by yaroslavborbat — DISMISSED** [review]

File: images/kube-api-proxy/pkg/rewriter/affinity.go (hunk 1 above)
Comment (diafour) [reply images/kube-api-proxy/pkg/rewriter/affinity.go]: Done

File: images/kube-api-proxy/pkg/rewriter/list.go (hunk 2 above)
Comment (diafour) [reply images/kube-api-proxy/pkg/rewriter/list.go]: Done for all such cases.

**Review by yaroslavborbat — DISMISSED** [review]

**Review by yaroslavborbat — APPROVED** [review]

**Review by LopatinDmitr — APPROVED** [review]

**Pull request merged**